doc/source/tutorial/stats/sampling.rst

```diff
@@ -0,0 +1,327 @@
+called `UNU.RAN <http://statmath.wu.ac.at/software/unuran/>`__ are used
+for speed and performance. Please look at
+`UNU.RAN's documentation <http://statmath.wu.ac.at/software/unuran/doc/unuran.html>`__
+for an in-depth explanation of these methods. It is heavily referred
+for writing this tutorial and the documentation of all the samplers.
+
+
+Introduction
+------------
+
+Random variate generation is the small field of research that deals with
+algorithms to generate random variates from various distributions. It is
+common to assume that a uniform random number generator is available.
+This is a program that produces a sequence of independent and identically
+distributed continuous U(0,1) random variates (i.e. uniform random variates
+on the interval (0,1)). Of course, real-world computers can never generate
+ideal random numbers and they cannot produce numbers of arbitrary precision
+but state-of-the-art uniform random number generators come close to this
+aim. Thus random variate generation deals with the problem of transforming
+such a sequence of U(0,1) random numbers into non-uniform random variates.
+
+Some methods to do that are:
+
+* The Inversion method: When the inverse :math:`F^{-1}` of the cumulative
+  distribution function is known, then random variate generation is easy.
+  We just generate a uniformly U(0,1) distributed random number U and
+  return :math:`X = F^{-1}(U)`. See
+  :class:`~NumericalInverseHermite`.
+* The Rejection Method: The rejection method, often called
+  acceptance-rejection method, has been suggested by John von Neumann in
+  1951. It involves computing an upper bound to the PDF (also called the
+  hat function) and using the inversion method to generate a random
+  variate, say Y, from this bound. Then a uniform random number can be
+  drawn between 0 to the value of the upper bound at Y. If this number
+  is less than the PDF at Y, return the sample otherwise reject it. See
+  :class:`~TransformedDensityRejection`.
+* The Ratio-of-Uniforms Method: This is a type of acceptance-rejection
+  method which is uses minimal bounding rectangles to construct the hat
+  function. See :func:`~rvs_ratio_uniforms`.
+* Inversion for Discrete Distributions: The difference compared to the
+  continuous case is that :math:`F` is now a step-function. To realize
+  this in a computer, a search algorithm is used, the simplest of which
+  is *sequential search*. A uniform random number is generated from
+  U(0, 1) and probabilities are summed until the cumulative probability
+  exceeds the uniform random number. The index at which this happens is
+  the required random variate and is returned.
+
+
+More details on these algorithms can be found in the `appendix of the UNU.RAN
+user manual <http://statmath.wu.ac.at/software/unuran/doc/unuran.html#RVG>`__.
+
+
+Basic concepts of the Interface
+-------------------------------
+
+Every sampler needs to be set up before one can start sampling from it.
```

Comment:
Should we say generator instead of sampler? The former term is used in UNU.RAN

Reply:
> Should we say generator instead of sampler?

I was also confused between the two naming conventions. So, I have renamed "sampler" to "generator" to follow one common convention.